ios-target/Networking/Services/TargetAPI.swift

```diff
@@ -14,25 +14,37 @@ class TargetAPI {
   
   class func getTargets(_ success: @escaping ([Target]) -> Void, failure: @escaping (_ error: Error) -> Void) {
     APIClient.request(.get, url: targetsUrl, success: { response, _ in
-      guard let targets = response["targets"] as? [[String: Any]], !targets.isEmpty else {
+      guard let targets = response["targets"] as? [[String: Any]] else {
           failure(App.error(domain: .parsing, localizedDescription: "Could not parse valid targets".localized))
           return
         }
         success(Target.parse(targets: targets))
     }, failure: failure)
   }
   
-  class func createTarget(_ target: Target, success: @escaping (_ response: Target) -> Void, failure: @escaping (_ error: Error) -> Void) {
+  class func createTarget(_ target: Target, success: @escaping (_ response: [String: Any]) -> Void, failure: @escaping (_ error: Error) -> Void) {
     let parameters = Target.buildParams(target: target)
     APIClient.request(.post, url: targetsUrl, params: parameters, success: { response, _ in
-      guard let target = response["target"] as? [String: Any],
-        !target.isEmpty,
-        let parsedTarget = try? JSONDecoder().decode(Target.self, from: target)
-      else {
+      if let target = response["target"] as? [String: Any],
+        let parsedTarget = try? JSONDecoder().decode(Target.self, from: target) {
+        
+        if let matchConversation = response["match_conversation"] as? [String: Any],
```

Comment:
I think we need a Match model to avoid this manual parsing everywhere and tracking of `topicId` and `matchUser`